#!/usr/bin/env python
PACKAGE = "aerial_robot_base"


from dynamic_reconfigure.parameter_generator_catkin import *
from aerial_robot_msgs.msg import DynamicReconfigureLevels
from math import pi

gen = ParameterGenerator()

gen.add("kalmanFilterFlag", bool_t, DynamicReconfigureLevels.RECONFIGURE_KALMAN_FILTER_FLAG, "Kalman Filter Flag", False)
gen.add("inputSigma", double_t, DynamicReconfigureLevels.RECONFIGURE_INPUT_SIGMA, "Input Sigma", 0.04, 0, 1)
gen.add("biasSigma", double_t, DynamicReconfigureLevels.RECONFIGURE_BIAS_SIGMA, "Bias Sigma", 0.0005, 0, 1)
gen.add("measureSigma", double_t, DynamicReconfigureLevels.RECONFIGURE_MEASURE_SIGMA, "Measure Sigma", 0.04, 0, 1)

exit(gen.generate(PACKAGE, "aerial_robot_base", "StateKalmanFilter"))


